vignettes/spatial_wwinference.Rmd

```diff
@@ -0,0 +1,2457 @@
+description: "Here we showcase how we can add a site to site correlations."
+author: "Christian O. Bernal Zelaya"
+date: "2024-09-13"
+output:
+  html_document:
+    fig_width: 15
+    fig_height: 12
+    number_sections: true
+    toc: true
+    toc_depth: 1
+pkgdown:
+  as_is: true
+vignette: >
+  %\VignetteIndexEntry{Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+```{r echo=FALSE}
+knitr::opts_chunk$set(dev = "svg")
+```
+
+<script type="text/javascript">
+$(document).ready(function() {
+
+  $chunks = $('.fold');
+
+  $chunks.each(function () {
+
+    // Add button to source code chunks
+    if ( $(this).hasClass('s') ) {
+      $('pre.r', this).prepend("<div class=\"showopt\">Hide Source</div><br style=\"line-height:22px;\"/>");
+      $('pre.r', this).children('code').removeClass('folded'); // Default to show code
+    }
+
+    // Add button to output chunks
+    if ( $(this).hasClass('o') ) {
+      $('pre:not(.r)', this).has('code').prepend("<div class=\"showopt\">Show Output</div><br style=\"line-height:22px;\"/>");
+      $('pre:not(.r)', this).children('code:not(r)').addClass('folded'); // Default to hide output
+
+      // Add button to plots
+      $(this).find('img').wrap('<pre class=\"plot\"></pre>');
+      $('pre.plot', this).prepend("<div class=\"showopt\">Show Plot</div><br style=\"line-height:22px;\"/>");
+      $('pre.plot', this).children('img').addClass('folded'); // Default to hide plots
+
+    }
+  });
+
+  // Hide all chunks when document is loaded
+  $('.folded').css('display', 'none');
+
+  // Function to toggle the visibility
+  $('.showopt').click(function() {
+    var label = $(this).html();
+    if (label.indexOf("Show") >= 0) {
+      $(this).html(label.replace("Show", "Hide"));
+    } else {
+      $(this).html(label.replace("Hide", "Show"));
+    }
+    $(this).siblings('code, img').slideToggle('fast', 'swing');
+  });
+});
+</script>
+
+<style>
+.showopt {
+  background-color: #004c93;
+  color: #FFFFFF;
+  width: 100px;
+  height: 20px;
+  text-align: center;
+  vertical-align: middle !important;
+  float: right;
+  font-family: sans-serif;
+  border-radius: 8px;
+  cursor: pointer; /* Add cursor pointer for better UX */
+}
+
+.showopt:hover {
+  background-color: #dfe4f2;
+  color: #004c93;
+}
+
+pre.plot {
+  background-color: white !important;
+}
+</style>
+
+# Start
+Here we present the way simulating and fitting with 3 separate models using the
+`wwinference` package.  We also compare the results a number of ways to see
+the value of using spatial components within the model.
+
+<div class="fold s">
+```{r warning=FALSE, message=FALSE}
+library(wwinference)
+library(dplyr)
+library(tibble)
+library(tidyr)
+library(ggplot2)
+library(patchwork)
+library(scoringutils)
+library(ggh4x) # for facet_grid2
+```
+
+
+# Generating Data
+
+We start by generating the data that our models require, using the
+`generate_simulation_data` function.  The presets that are required to use the
```

Comment:
```suggestion
`generate_simulation_data()` function.  The presets that are required to use the
```

I think this will link to the docs (and should be done everywhere).